File: MiniProtobufClient/MoveProcessor.cpp
#include "MoveProcessor.h"


using namespace BaseCmd;
MoveProcessor::MoveProcessor()
{
}

MoveProcessor::~MoveProcessor()
{
}

void MoveProcessor::Process(std::shared_ptr<MessageRoute> spRoute, std::shared_ptr<google::protobuf::Message> spMsg)
{
	rspMove* p_rsp = reinterpret_cast<rspMove*>(spMsg.get());

	std::string error = p_rsp->error();
	std::cout << "move rsp:" << error << std::endl;

}

File: AndroidProj/app/src/main/java/com/dd/androidprotobuf/MoveProcessor.java
package com.dd.androidprotobuf;

import android.content.Intent;

import com.google.protobuf.GeneratedMessageLite;

import BaseCmd.Cmd;
import BaseCmd.Cmd.rspMove;

public class MoveProcessor extends  BaseProcessor {

    @Override
    void Process(MessageRoute route, GeneratedMessageLite msg)
    {
        rspMove rsp = (rspMove)msg;
        rsp.getError();


        Intent i = new Intent(BaseProcessor.MessageFilter);
        i.putExtra(Intent.EXTRA_TEXT,rsp.getError());
        MainActivity.Instance.sendBroadcast(i);
    }
}
